File: backend/OlivePlatform/OlivePlatform.Domain/Entities/Harvest.cs
using OlivePlatform.Domain.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace OlivePlatform.Domain.Entities;

[Table("harvests")]
public class Harvest
{
    [Column("id")]
    public int Id { get; set; }

    [Column("reference")]
    public string Reference { get; set; } = null!;

    [Column("plot_id")]
    public int PlotId { get; set; }

    [Column("harvest_date")]
    public DateOnly HarvestDate { get; set; }

    [Column("quantity_kg")]
    public decimal QuantityKg { get; set; }

    [Column("notes")]
    public string? Notes { get; set; }

    [Column("variety_id")]
    public int VarietyId { get; set; }

    [Column("start_time")]
    public DateTime? StartTime { get; set; }

    [Column("end_time")]
    public DateTime? EndTime { get; set; }

    [Column("harvested_trees")]
    public int? HarvestedTrees { get; set; }

    [Column("planned_trees")]
    public int? PlannedTrees { get; set; }


    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("status")]
    public ProductionStatus Status { get; set; }

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }
}

File: frontend/olive-platform/packages/core/src/features/harvests/domain/entities/Harvest.ts
import type { ProductionStatus } from "../../../production/domain/entities/ProductionStatus";
import type { OliveVarieties } from "../../../shared/entities/OliveVarieties";

export interface Harvest {
  id: number;
  reference: string;
  plotId: number;
  variety: OliveVarieties;
  harvestedTrees: number;
  plannedTrees: number;
  harvestDate: string;
  quantityKg: number | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string | null;
  status: ProductionStatus;
  startTime: string | null;
  endTime: string | null;
}
